### DISCUSSION TITLE
How are the Github actions minutes caluculated ?

### DISCUSSION THREAD

**Discussion by davidmiller9999** [machulav/ec2-github-runner#124] [Q&A] [answered]
- While the actions are running on an hosted ec2 instance do we get GitHub actions minutes charged ? 
- The reason i choose hosted runners is my actions takes longer some even hours, Since we only have 2000 free minutes and if action minutes gets consumed while it's waiting for the self hosted runner task to be completed and stop the runner after. what would be the alternative ?

**Comment by machulav** [comment, marked as answer]
Your use case is described on the main page of this repository under the "Save costs" header: 
https://github.com/machulav/ec2-github-runner#save-costs

You can also find a link to the official GitHub docs, which explains it in more detail.

TLDR: GitHub doe not charge for the self-hosted runner time.
